### PR TITLE
fix(resw): Adjust resources resolution

### PR DESCRIPTION
**Pull request by jeromelaban**
GitHub Issue (If applicable): closes https://github.com/unoplatform/uno/issues/10812

## PR Type

What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?

Fixes the resolution of resw-defined resources when located in a separate assemby.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs [hunk 1]
```diff
@@ -382,6 +384,14 @@ private string GetSourceLink(Uno.Roslyn.MSBuildItem projectItemInstance)
 			}
 		}
 
+		private void TryGenerateUnoResourcesKeyAttribute(ImmutableHashSet<string> resourceKeys)
+		{
+			if (resourceKeys.Any())
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs:389]:
```suggestion
			if (!resourceKeys.IsEmpty)
```

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 2]
```diff
@@ -733,80 +736,56 @@ private void GenerateResourceLoader(IndentedStringBuilder writer)
...
+                else
+                {
+                    throw new InvalidOperationException($"Unsupported resource type for {reference.Display} ({reference.GetType()})");
+                }
+            }
+		}
+
+        private void BuildResourceLoaderFromAssembly(IndentedStringBuilder writer, IAssemblySymbol assembly)
+        {
+			if (assembly.GetAllAttributes().Any(a =>
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:754]: This should be sufficient. `GetAllAttributes` can only behave differently for `INamedTypeSymbol`s

```suggestion
			if (assembly.GetAttributes().Any(a =>
```

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 3]
```diff
@@ -733,80 +736,56 @@ private void GenerateResourceLoader(IndentedStringBuilder writer)
...
+				SymbolEqualityComparer.Default.Equals(a.AttributeClass, _assemblyMetadataSymbol)
+				&& a.ConstructorArguments.Length == 2
+				&& a.ConstructorArguments[0].Value?.ToString() == "UnoHasLocalizationResources"
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:757]:
```suggestion
				&& a.ConstructorArguments[0].Value is "UnoHasLocalizationResources"
```

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs [hunk 4]
```diff
@@ -733,80 +736,56 @@ private void GenerateResourceLoader(IndentedStringBuilder writer)
...
+				&& (a.ConstructorArguments[1].Value?.ToString().Equals("True", StringComparison.OrdinalIgnoreCase) ?? false)))
+			{
+				if (_isWasm)
 				{
-					if (reference is PortableExecutableReference per && File.Exists(per.FilePath))
-					{
-						return per.FilePath;
-					}
-					else if (File.Exists(reference.Display))
+					var anchorType = assembly.Modules.First().GlobalNamespace.GetNamespaceTypes().FirstOrDefault(s => s.IsLocallyPublic(_metadataHelper.Compilation.Assembly.Modules.First()));
```
Comment (Youssef1313) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:762]: This might be expensive.

Comment (jeromelaban) [reply src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs:762]: it's taking about 1.0ms on `Uno.UI.RemoteControl`, then 0.2ms `Uno.UI.RuntimeTests`. Most of the lookup is don through enumerables provided by `GetNamespaceTypes()`, so until we find a better way to handle this (this is a mono-wasm limitation), we'll need to live with it... it was previously done through a cecil lookup that was not particularly efficient either:

https://github.com/unoplatform/uno/blob/3cb7e5f97a9226ba3d0b53756e61a0d94832169c/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs#L796

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs [hunk 5]
```diff
@@ -378,6 +380,14 @@ private string GetSourceLink(Uno.Roslyn.MSBuildItem projectItemInstance)
 			}
 		}
 
+		private void TryGenerateUnoResourcesKeyAttribute(ImmutableHashSet<string> resourceKeys)
+		{
+			if (resourceKeys.IsEmpty)
+			{
+				_generatorContext.AddSource("LocalizationResources", "[assembly: global::System.Reflection.AssemblyMetadata(\"UnoHasLocalizationResources\", \"True\")]");
+			}
```
Comment (carldebilly) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs:388]: It seems counter-intuitive that having now resources (is empty) is setting a flag telling there is localized resources...

**Review by carldebilly — APPROVED** [review]

**Review by dr1rrb — APPROVED** [review]

**Review by spouliot — APPROVED** [review]

**Pull request merged**